sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs

```diff
@@ -42,28 +36,26 @@ internal sealed class EventHubListener : IListener, IEventProcessorFactory, ISca
 
         public EventHubListener(
             string functionId,
-            string eventHubName,
-            string consumerGroup,
-            string connectionString,
-            string storageConnectionString,
             ITriggeredFunctionExecutor executor,
             EventProcessorHost eventProcessorHost,
             bool singleDispatch,
             EventHubOptions options,
             ILogger logger,
             BlobContainerClient blobContainer = null)
         {
-            _functionId = functionId;
-            _eventHubName = eventHubName;
-            _consumerGroup = consumerGroup;
-            _connectionString = connectionString;
-            _storageConnectionString = storageConnectionString;
             _executor = executor;
             _eventProcessorHost = eventProcessorHost;
             _singleDispatch = singleDispatch;
             _options = options;
             _logger = logger;
-            _scaleMonitor = new Lazy<EventHubsScaleMonitor>(() => new EventHubsScaleMonitor(_functionId, _eventHubName, _consumerGroup, _connectionString, _storageConnectionString, _logger, blobContainer));
+            _scaleMonitor = new Lazy<EventHubsScaleMonitor>(() => new EventHubsScaleMonitor(
```

Comment:
is there a benefit to it?

Reply:
As I understand it, a minor one.   Instead of grabbing a primitive to ensure that only one thread can create the value, it will allow multiple threads to race for it and keep just the first via atomic swap.